src/components/SettingsDrawer.jsx

```diff
@@ -0,0 +1,194 @@
+import { useState, useEffect } from "react";
+
+import {
+  Brightness4 as Brightness4Icon,
+  Brightness7 as Brightness7Icon,
+  Close as CloseIcon,
+  Language as LanguageIcon,
+  Business as BusinessIcon,
+  SettingsBrightness as SettingsBrightnessIcon
+} from "@mui/icons-material";
+import PaletteIcon from "@mui/icons-material/Palette";
+import {
+  Drawer,
+  Paper,
+  Box,
+  List,
+  ListItem,
+  ListItemIcon,
+  ListItemText,
+  Divider,
+  Select,
+  MenuItem,
+  Typography,
+  Button,
+  IconButton
+} from "@mui/material";
+import PropTypes from "prop-types";
+import { useNavigate } from "react-router-dom";
+
+import { useGlobalSetting } from "~/contexts/GlobalSettingProvider";
+import { getSetting, setSetting } from "~/hooks/settingsService";
+
+const SettingsDrawer = ({ open, handleClose }) => {
+  const [themeMode, setThemeMode] = useState(() => getSetting("themeMode", "light"));
+  const [language, setLanguage] = useState(() => getSetting("language", "english"));
+  const [themeColor, setThemeColor] = useState(() => getSetting("themeColor", ""));
+  const [selectedTenant, setSelectedTenant] = useState(() => getSetting("selectedTenant", "1801"));
```

Comment:
this implementation is not correct, Please refactor it

Reply:
why is this not correct? and what is the recommended way?

Reply:
as you are using this as utility funtion best way should directly get all the local storage value to a single varaible or state and have a updater function to update the specific keys

Reply:
> this implementation is not correct, Please refactor it
why is this not correct? and what is the recommended way?